packages/core/src/new-api/internal/new-execution/reducers/deployment-state-reducer.ts

```diff
@@ -0,0 +1,92 @@
+import { IgnitionError } from "../../../../errors";
+import { assertIgnitionInvariant } from "../../utils/assertions";
+import { DeploymentState, ExecutionStateMap } from "../types/deployment-state";
+import { ExecutionSateType, ExecutionState } from "../types/execution-state";
+import {
+  JournalMessage,
+  JournalMessageType,
+  RunStartMessage,
+} from "../types/messages";
+
+import { deploymentExecutionStateReducer } from "./deployment-execution-state-reducer";
+import { assertUnknownAction } from "./utils";
+
+const initialState: DeploymentState = {
+  chainId: 0,
+  executionStates: {},
+};
+
+export function deploymentStateReducer(
+  state: DeploymentState = initialState,
+  action?: JournalMessage
+): DeploymentState {
+  if (action === undefined) {
+    return {
+      chainId: 0,
+      executionStates: {},
+    };
+  }
+
+  switch (action.type) {
+    case JournalMessageType.RUN_START:
+      return {
+        ...state,
+        chainId: action.chainId,
+      };
+    case JournalMessageType.DEPLOYMENT_EXECUTION_STATE_INITIALIZE:
+    case JournalMessageType.NETWORK_INTERACTION_REQUEST:
+    case JournalMessageType.TRANSACTION_SEND:
+      return {
+        ...state,
+        executionStates: executionStatesReducer(state.executionStates, action),
+      };
+    default:
+      return assertUnknownAction(action);
+  }
+}
+
+function executionStatesReducer(
+  state: ExecutionStateMap = {},
+  action: Exclude<JournalMessage, RunStartMessage>
```

Comment:
I found something useful for the reducers. You don't need to do this exclude thing, which will break when you add more things.

Instead, if you first write the function call to `executionStatesReducer`, and then ask vscode to generate it, it has tighter types. 

That autogenerated code normally looks ugly, but if you add `import * as Messages from "../types/messages";` they look good.

Also, those tighter types eliminate the need to do `default: assert unknown`.